core/src/main/scala/org/scalacheck/ops/time/ImplicitJavaTimeGenerators.scala

```diff
@@ -30,24 +30,42 @@ trait ImplicitJavaTimeGenerators {
   implicit val arbInstant: Arbitrary[Instant] = {
     Arbitrary {
       for {
-        millis <- chooseNum(0L, Instant.MAX.getEpochSecond)
-        nanos <- chooseNum(0, Instant.MAX.getNano)
+        millis <- chooseNum(Instant.MIN.getEpochSecond, Instant.MAX.getEpochSecond)
+        nanos <- chooseNum(Instant.MIN.getNano, Instant.MAX.getNano)
       } yield {
         Instant.ofEpochMilli(millis).plusNanos(nanos)
       }
     }
   }
 
-  implicit val arbLocalDateTime: Arbitrary[LocalDateTime] = {
+  implicit val arbLocalDate: Arbitrary[LocalDate] = {
     Arbitrary {
       for {
-        instant <- arbInstant.arbitrary
-      } yield LocalDateTime.from(instant)
+        epochDay <- chooseNum(LocalDate.MIN.toEpochDay, LocalDate.MAX.toEpochDay)
+      } yield LocalDate.ofEpochDay(epochDay)
+    }
+  }
+
+  implicit val arbLocalTime: Arbitrary[LocalTime] = {
+    Arbitrary {
+      for {
+        nanoOfDay <- chooseNum(LocalTime.MIN.toNanoOfDay, LocalTime.MAX.toNanoOfDay)
+      } yield LocalTime.ofNanoOfDay(nanoOfDay)
+    }
+  }
+
+  implicit def arbLocalDateTime(implicit arbZoneOffset: Arbitrary[ZoneOffset]): Arbitrary[LocalDateTime] = {
+    Arbitrary {
+      for {
+        offset <- arbZoneOffset.arbitrary
+        seconds <- chooseNum(LocalDateTime.MIN.toEpochSecond(offset), LocalDateTime.MAX.toEpochSecond(offset))
+        nanos <- chooseNum(LocalDateTime.MIN.getNano, LocalDateTime.MAX.getNano)
+      } yield LocalDateTime.ofEpochSecond(seconds, nanos, offset)
```

Comment:
Do you need `Arbitrary[ZoneOffset]`, why changes if `ZoneOffset.UTC` is used here?

Reply:
`LocalDateTime` doesn't actually store the timezone, it just uses it to calculate the given time into a local time equivalent based on the offset. You're right though, if it is an arbitrary datetime, I don't really need to use an arbitrary timezone offset...

I guess I was worried about making any assumptions for testing purposes. I don't think it harms anything to use an arbitrary timezone offset. I'll look into this a bit more though.